includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -417,7 +417,7 @@ public function get_product_data() {
 			'pending' => true,
 		];
 
-		$data['requestShipping'] = ( wc_shipping_enabled() && $product->needs_shipping() );
+		$data['requestShipping'] = ( wc_shipping_enabled() && $product->needs_shipping() && 0 !== wc_get_shipping_method_count( true ) );
```

Comment:
Here I'm using what WC Cart's `needs_shipping` method is using 

https://github.com/Automattic/woocommerce/blob/77127f93af6edd0384810a3e440fbddcbea5a751/includes/class-wc-cart.php#L1526